File: leetcode/977. Squares of a Sorted Array.cpp
#include <algorithm>
class Solution {
public:
    vector<int> sortedSquares(vector<int>& A) {
        vector<int> res(A.size());
        int i = 0;
        int j = A.size() - 1;
        int idx = A.size() - 1;
        while(i <= j){
            if(abs(A[i]) > abs(A[j])){
                res[idx] = A[i] * A[i];
                ++i;
            } else {
                res[idx] = A[j] * A[j];
                --j;
            }
            --idx;
        }
        return res;
    }
};

File: leetcode/977. Squares of a Sorted Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        i = 0
        j = len(nums)-1
        res = []
        while i <= j:
            if abs(nums[i]) < abs(nums[j]):
                res.append(nums[j]**2)
                j -= 1
            else:
                res.append(nums[i]**2)
                i += 1
        res.reverse()
        return res
